### ISSUE TITLE
Having a directory called "sdk" next to the actual sdk causes errors

### ISSUE THREAD

**Issue by jakemac53** [Dart-Code/Dart-Code#421]
Bug report text is here https://gist.github.com/jakemac53/a5b624254b9680481a0a301ba3d635bc.

You can repro this by adding a directory called `sdk` next to your main sdk (for me they were both in my home dir).

You can work around this by renaming the `sdk` dir.

**Comment by DanTup** [comment]
I think this might be a bug in the SDK (the error comes from `PluginWatcher._getSdkPath`)  so may need to go to [dart-lang/sdk](https://github.com/dart-lang/sdk) (@bwilkerson should be able to confirm).

**Comment by bwilkerson** [comment]
Yes, the exception is definitely a server bug. I'm not sure what caused it, but it would be useful to know which of the two directories server was told to use as the SDK and whether they are both configured as expected (with `.libraries` files, etc.).

**Comment by DanTup** [comment]
> it would be useful to know which of the two directories server was told to use as the SDK

IIRC Dart Code doesn't tell the server where the SDK is; it just locates an SDK then uses it for both the VM being launched and the server snapshot:

```
	const analyzerPath = config.analyzerPath || path.join(sdks.dart, util.analyzerPath);
	analyzer = new Analyzer(path.join(sdks.dart, util.dartVMPath), analyzerPath);
```

`sdks.dart` is the path to the SDK we found and `util.analyzerPath` is the relative path to the server snapshot.

I don't understand the comment about the `.libraries` file (I don't know what it is!), though not sure if that's aimed at me or Jake?

**Comment by jakemac53** [comment]
I think the `.libraries` file is an implementation detail? It is something which ships with the built sdk I believe, and it fails to load since analysis server thinks the sdk is in a directory that is not actually the sdk.

**Comment by bwilkerson** [comment]
> I think the .libraries file is an implementation detail?

The `.libraries` file is how analyzer/server knows which `dart:` libraries are defined and where to find them. So, yes, it's an implementation detail, but one that's essential.

> ... analysis server thinks the sdk is in a directory that is not actually the sdk.

Yep. What I need to do is figure out why it's picked the wrong location for the sdk.

**Comment by DanTup** [comment]
I can also repro this quite easily:

- SDK 1.24.2 is at `M:\Apps\Dart\Latest`
- Create empty folder at `M:\Apps\Dart\sdk`
- Run `M:\Apps\Dart\Latest\bin\dart.exe M:\Apps\Dart\Latest\bin\snapshots\analysis_server.dart.snapshot`
- Send `{"id":"2","method":"analysis.setAnalysisRoots","params":{"included":["c:\\Users\\danny\\Desktop\\DartSample"],"excluded":[]}}`

I've raised https://github.com/dart-lang/sdk/issues/30651 since I don't believe it's Dart Code specific and presume @bwilkerson will be able to repro fairly easily.
